cdap-ui/cypress/integration/httpexecutor.requesthistorytab.spec.ts

```diff
@@ -0,0 +1,253 @@
+      cy.get('[data-cy*="request-row"]').should('have.length', 1);
+
+      // None of the requests include searchText3 in their path
+      const searchText3 = 'hellooo';
+      cy.get(dataCy('request-search-input'))
+        .click()
+        .focused()
+        .clear()
+        .type(searchText3);
+      cy.get('[data-cy*="request-row"]').should('have.length', 0);
+
+      // Clear out all the search text
+      cy.get(dataCy('request-search-input'))
+        .click()
+        .focused()
+        .clear();
+    });
+
+    it('should add a request to requestHistoryTab', () => {
+      // Should add a new request when the save mode is off
+      cy.get(dataCy('save-mode-btn')).click(); // turn off the save mode
+      const newRequest1 = {
+        method: 'DELETE',
+        path: 'https://new-request-1.com',
+      };
+      // Attempt to add a new request
+      cy.get(dataCy('request-method-selector')).select(newRequest1.method);
+      cy.get(dataCy('request-path-input'))
+        .clear()
+        .click()
+        .type(newRequest1.path);
+      cy.get(dataCy('send-btn')).click();
+      // Validate a new request has NOT been added to requestHistoryTab
+      cy.get('[data-cy*="request-row"]').should(
+        'have.length',
+        Object.keys(MOCK_LOCAL_STORAGE).length
+      );
+
+      // Should add a new request when the save mode is on
+      cy.get(dataCy('save-mode-btn')).click(); // turn on the save mode
+      const newRequest2 = {
+        method: 'POST',
+        path: 'https://new-request-2.com',
+        body: 'hello4',
+      };
+      // Attempt to add a new request
+      cy.get(dataCy('request-method-selector')).select(newRequest2.method);
+      cy.get(dataCy('request-path-input'))
+        .clear()
+        .click()
+        .type(newRequest2.path);
+      cy.get(dataCy('request-body'))
+        .clear()
+        .click()
+        .type(newRequest2.body);
+      cy.get(dataCy('send-btn')).click();
+      // Validate a new request has been added to requestHistoryTab
+      cy.get('[data-cy*="request-row"]').should(
+        'have.length',
+        Object.keys(MOCK_LOCAL_STORAGE).length + 1
+      );
+    });
+
+    it('should delete a request from RequestHistoryTab', () => {
+      // Delete the latest request from RequestHistoryTab
+      cy.get('[data-cy*="request-row"]').within(() => {
+        // Since delete-icon is a hidden element, we need click({force: true})
+        cy.get(dataCy('delete-icon'))
+          .first() // the first element has the latest request
+          .invoke('show')
+          .click({ force: true });
+      });
+
+      // Check whether a delete dialog has been opened
+      cy.get(dataCy('confirm-dialog')).should('exist');
+
+      // Confirm delete
+      cy.get(dataCy('Delete')).click();
+
+      // Validate the request has been deleted from requestHistoryTab
+      cy.get('[data-cy*="request-row"]').should(
+        'have.length',
+        Object.keys(MOCK_LOCAL_STORAGE).length - 1
```

Comment:
Shouldn't this be MOCK_LOCAL_STORAGE (without the minus one) since the previous test added another request?

Reply:
Done!